bin/beeline

```diff
@@ -28,5 +28,7 @@ if [ -z "${SPARK_HOME}" ]; then
   source "$(dirname "$0")"/find-spark-home
 fi
 
+. "${SPARK_HOME}"/bin/load-spark-env.sh
+
 CLASS="org.apache.hive.beeline.BeeLine"
-exec "${SPARK_HOME}/bin/spark-class" $CLASS "$@"
+exec "${SPARK_HOME}/bin/spark-class" $SPARK_SUBMIT_OPTS $CLASS "$@"
```

Comment:
It executes Hive's beeline. Does it respect Spark summit options?

Reply:
Hi Hyukjin, some discussions about the krb5.conf location before is here https://issues.apache.org/jira/browse/SPARK-12050, but in this case, this does not work. From the discussion, people saying we can use SPARK_SUBMIT_OPTS to include the location by  `-Djava.security.krb5.conf=/etc/krb5.conf-custom`.

Reply:
Was the JIRA for Hive beeline? I'm  saying its Hive's, not Spark's

Reply:
From my understanding, It should not be hive beeline's issue. Its an environment variable. This might not be the best solution, but it does solve the issue here. I have test hive beeline also, it works because it is able to read HADOOP_OPS. But somehow for spark, in spark-class script, whatever specified in the SPARK_SUBMIT_OPS is not passed to execution. So here I have to pass it in for spark beeline.

Reply:
Let's avoid a bandaid fix here. Can we make `HADOOP_OPS` work with the beeline in Spark?

Reply:
I took a look of how hive's beeline works, it looks like the suggested fix for spark's beeline aligns well with that.
Also, spark's beeline actually calls "spark-class" (and some other code call it as well), after tracing them one by one, the suggested fix seems to be the safest way as well.

Reply:
@kdzhao, can you show some code pointers? I still think it's odd that `SPARK_SUBMIT_OPTS` should take an effect in Hive's beeline.

Reply:
I think we misread each other. What I want to say is, in hive, looks like its beeline command is just call to hive with different parameters:
https://github.com/apache/hive/blob/branch-1.2/bin/beeline
https://github.com/apache/hive/blob/branch-1.2/bin/hive
Agree that hive's beeline doesn't read spark parameter, and I would assume it reads its own (I saw "HADOOP_CLIENT_OPTS" etc in above script).
Now back to spark, agree with you that fixing it in spark-class might cover more cases. On another side, so far only the beeline has this issue, so an easy fix on the beeline script also makes sense as a stopgap.